test/cluster/mzcompose.py

```diff
@@ -1901,3 +1902,77 @@ def workflow_test_clusterd_death_detection(c: Composition) -> None:
         "error reading a body from connection: stream closed because of a broken pipe"
         in envd.stdout
     )
+
+
+def workflow_test_replica_dataflow_metrics(c: Composition) -> None:
+    """Test dataflow metrics exposed by replicas."""
+
+    c.down(destroy_volumes=True)
+    c.up("materialized")
+    c.up("clusterd1")
+
+    def fetch_metrics(name: str, collection_id: str) -> List[float]:
+        metrics = c.exec(
+            "clusterd1", "curl", "localhost:6878/metrics", capture=True
+        ).stdout
+
+        values = []
+        for metric in metrics.splitlines():
+            if metric.startswith(name) and f'collection_id="{collection_id}"' in metric:
+                value = float(metric.split()[1])
+                values.append(value)
+
+        return values
```

Comment:
Nit: You could allow re-using the metrics by returning a map string -> string here, and then lookup individual values. This should make the test faster, although I can't say by how much.

Reply:
Probably doesn't make a huge difference, since fetching metrics is just a local HTTP request, but just to be sure I've changed to code to make sure we fetch new metrics only when needed.